packages/astro/src/core/config/settings.ts

```diff
@@ -134,3 +145,63 @@ export async function createSettings(config: AstroConfig, cwd?: string): Promise
 
 	return settings;
 }
+
+function validateTsconfig(settings: AstroSettings, logger: Logger, rawConfig: TSConfig) {
```

Comment:
Personally I'm not sure about validating the tsconfig ourselves. Tsconfig can come in a lot of kinds and configurations, and it may not always be possible to enforce `extends` to exist, nor 100% sure that the `include` and `exclude` are not set. For example, here's Vite's setup: https://github.com/vitejs/vite/blob/cfb621e7a5a3e24d710a9af156e6855e73caf891/packages/create-vite/template-react-ts/tsconfig.json#L1-L8

The feature is experimental so I don't mind if we want to try this, but I think it would be better to focus on improving the docs on this instead.

Reply:
The goal here was to help the user. Like if it comes out of experimental in 5.0, users may wonder why their types stopped working all of sudden. Explaining it in the terminal as early as possible sounds like the perfect time to me but happy to revisit.

Regarding vite's setup, I'm not sure what impact references have

Reply:
My concern is that tsconfig can be wildly different based on how the user wants to configure it. e.g. maybe they want to have `tsconfig.json -> tsconfig.base.json -> .astro/tsconfig.json`, or if they use [project references](https://www.typescriptlang.org/docs/handbook/project-references.html) and the extends is somewhere else like `tsconfig.app.json`. 

So I don't think it's in our territory to validate and ensure they have this specific shape. Also, will this be a breaking change in Astro 5 when enabled by default? Will `env.d.ts` not work anymore?

Reply:
Ah I see what you mean. As for `env.d.ts`, the plan is to stop writing it in favor of just the tsconfig

Reply:
I am with @bluwy here. This could get hairy if you step into a corporate project where there are many shapes fo configuration files. Let's remember that there's also `jsconfig.json` too.

So yeah... I suggest to not do it